tensorflow/contrib/tensorrt/python/trt_convert.py

```diff
@@ -121,41 +116,41 @@ def py3string(inp):
     to_bytes = py3bytes
     to_string = py3string
 
-  out_names = []
-  for i in outputs:
-    if isinstance(i, ops.Tensor):
-      out_names.append(to_bytes(i.name))
-    else:
-      out_names.append(to_bytes(i))
-
-  input_graph_def_str = input_graph_def.SerializeToString()
-
-  # TODO(sami): Fix this when we can return status from C++ library
-  # There is a problem with the TF internal library setup that doesn't
-  # allow us to return a status object from C++.  Thus we return a
-  # pair or strings where first one is encoded status and the second
-  # one is the transformed graphs protobuf string.
-  out = trt_convert(input_graph_def_str, out_names, max_batch_size,
-                    max_workspace_size_bytes, mode, minimum_segment_size,
-                    is_dynamic_op, maximum_cached_engines,
-                    cached_engine_batches)
-  status = to_string(out[0])
-  output_graph_def_string = out[1]
-  del input_graph_def_str  # Save some memory
-  if len(status) < 2:
-    raise _impl.UnknownError(None, None, status)
-  if status[:2] != "OK":
-    msg = status.split(";")
-    if len(msg) == 1:
-      raise RuntimeError("Status message is malformed {}".format(status))
-    # pylint: disable=protected-access
-    raise _impl._make_specific_exception(None, None, ";".join(msg[1:]),
-                                         int(msg[0]))
-    # pylint: enable=protected-access
-  output_graph_def = graph_pb2.GraphDef()
-  output_graph_def.ParseFromString(output_graph_def_string)
-  del output_graph_def_string  # Save some memory
-  return output_graph_def
+  # Create MetaGraphDef
+  graph = ops.Graph()
+  with graph.as_default():
+    importer.import_graph_def(input_graph_def, name="")
+  meta_graph = saver.export_meta_graph(
+      graph_def=graph.as_graph_def(), graph=graph)
+  if outputs:
+    output_collection = meta_graph_pb2.CollectionDef()
+    output_list = output_collection.node_list.value
+    for i in outputs:
+      if isinstance(i, ops.Tensor):
+        output_list.append(to_bytes(i.name))
+      else:
+        output_list.append(to_bytes(i))
+    meta_graph.collection_def["train_op"].CopyFrom(output_collection)
+
+  # Create RewriterConfig.
+  rewriter_cfg = rewriter_config_pb2.RewriterConfig()
+  rewriter_cfg.optimizers.extend(["constfold", "layout"])
+  optimizer = rewriter_cfg.custom_optimizers.add()
+  optimizer.name = "TensorRTOptimizer"
```

Comment:
Is this py3 safe?

Reply:
Yes, I tested that this works on both py2 and py3.